cmd/panfs.go

```diff
@@ -612,7 +640,23 @@ func (fs *PANFSObjects) ListBuckets(ctx context.Context, opts BucketOptions) ([]
 		return nil, err
 	}
 
-	entries, err := readDirWithOpts(fs.fsPath, readDirOpts{count: -1, followDirSymlink: true})
+	var entries []string
+	var err error
+
+	if fs.panclient != nil {
+		entries, err = fs.panclient.GetObjectsList(panfsBucketListPrefix)
+
+		if err == nil {
+			// The entries will have a prefix. In order to get the
+			// names of the buffers, we need to remove the prefixes.
+			for idx, entry := range entries {
+				entries[idx] = strings.TrimPrefix(entry, panfsBucketListPrefix)
+			}
+		}
+	} else {
+		entries, err = readDirWithOpts(fs.fsPath, readDirOpts{count: -1, followDirSymlink: true})
```

Comment:
`fs.fsPath` variable will be completely removed in future (or at least will be used to store some data usage statistics as a default path) so I think we only should fetch the bucket list from the configuration agent and not reading the filesystem.